src/components/Onboarding/Screens/Apps/VotingSettings.js

```diff
@@ -1,19 +1,265 @@
-import React from 'react'
+import React, {
+  Fragment,
+  useCallback,
+  useReducer,
+  useRef,
+  useState,
+} from 'react'
+import { GU, Help } from '@1hive/1hive-ui'
 import { useOnboardingState } from '@providers/Onboarding'
 import Navigation from '../../Navigation'
+import {
+  TimeParameterPanel,
+  Header,
+  PercentageField,
+  Carousel,
+} from '../../kit'
+import { DAY_IN_SECONDS } from '@/utils/kit-utils'
 
-function VotingSettings({ title }) {
-  const { onBack, onNext, step, steps } = useOnboardingState()
+const DEFAULT_SUPPORT = 50
+const DEFAULT_MIN_ACCEPTANCE_QUORUM = 10
+const DEFAULT_DURATION = DAY_IN_SECONDS * 5
+const DEFAULT_DELEGATED_VOTING_PERIOD = DAY_IN_SECONDS * 2
+const DEFAULT_QUITE_ENDING_PERIOD = DAY_IN_SECONDS * 1
+const DEFAULT_QUITE_ENDING_EXTENSION_PERIOD = DAY_IN_SECONDS * 0.5
+const DEFAULT_EXECUTION_DELAY_PERIOD = DAY_IN_SECONDS * 1
+
+const reduceFields = (fields, [field, value]) => {
+  switch (field) {
+    case 'support':
+      return {
+        ...fields,
+        support: value,
+        quorum: Math.min(fields.quorum, value),
+      }
+    case 'quorum':
+      return {
+        ...fields,
+        quorum: value,
+        support: Math.max(fields.support, value),
+      }
+    case 'duration':
+      return {
+        ...fields,
+        duration: value,
+      }
+    case 'delegatedVotingPeriod':
+      return {
+        ...fields,
+        delegatedVotingPeriod: value,
+      }
+    case 'quiteEndingPeriod':
+      return {
+        ...fields,
+        quiteEndingPeriod: value,
+      }
+    case 'quiteEndingExtensionPeriod':
+      return {
+        ...fields,
+        quiteEndingExtensionPeriod: value,
+      }
+    case 'executionDelayPeriod':
+      return {
+        ...fields,
+        executionDelayPeriod: value,
+      }
+    default:
+      return fields
+  }
+}
+
+function VotingSettings() {
+  const [isLastItemChecked, setIsLastItemChecked] = useState(false)
+  const { onBack, onNext, onConfigChange, step, steps } = useOnboardingState()
+  const [
+    {
+      support,
+      quorum,
+      duration,
+      delegatedVotingPeriod,
+      quiteEndingPeriod,
+      quiteEndingExtensionPeriod,
+      executionDelayPeriod,
+    },
+    updateField,
+  ] = useReducer(reduceFields, {
+    support: DEFAULT_SUPPORT,
+    quorum: DEFAULT_MIN_ACCEPTANCE_QUORUM,
+    duration: DEFAULT_DURATION,
+    delegatedVotingPeriod: DEFAULT_DELEGATED_VOTING_PERIOD,
+    quiteEndingPeriod: DEFAULT_QUITE_ENDING_PERIOD,
+    quiteEndingExtensionPeriod: DEFAULT_QUITE_ENDING_EXTENSION_PERIOD,
+    executionDelayPeriod: DEFAULT_EXECUTION_DELAY_PERIOD,
```

Comment:
We should use the `config.voting` that comes for the onboarding state provider (and set all this defaults there). Otherwise when coming back to this screen, the values would be reseted. Also was wondering if we should use the same naming convention for the attributes

Reply:
Yes, we should use the same naming convention.